### PR TITLE
Use reusable workflows

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**

### CODE REVIEW

**Comment by carlcsaposs-canonical** [comment]
TODO: switch from dpw branch to release

blocked by
https://github.com/canonical/data-platform-workflows/pull/143
https://github.com/canonical/data-platform-workflows/pull/144

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  release:
+    name: Release snap
+    needs:
+      - build
+    uses: canonical/data-platform-workflows/.github/workflows/release_snap.yaml@build-arm-4
+    with:
+      channel: latest/edge
+      artifact-prefix: ${{ needs.build.outputs.artifact-prefix }}
+    secrets:
+      snap-store-token: ${{ secrets.SNAP_STORE_TOKEN }}
```
Comment (carlcsaposs-canonical) [inline .github/workflows/release.yaml:28]: TODO: update GH secret or remove `candid` auth from reusable workflow

pending response from @taurus-forever

**Comment by carlcsaposs-canonical** [comment]
tested on
https://github.com/canonical-test4/charmed-postgresql-snap/actions/runs/8080081067
https://github.com/canonical-test4/charmed-postgresql-snap/actions/runs/8080766405?pr=3

**Review by dragomirp — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (taurus-forever) [reply .github/workflows/release.yaml:28]: SNAP_STORE_TOKEN inserted here.

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**